### PR TITLE
fix(hybrid-cloud): Fix client config for superuser

### PR DESCRIPTION
**Pull request by dashed**
I noticed the client config is deleting the `activeorg` from the request session object when the current user is in superuser mode. A side effect of this that we're not properly propagating the accessed org in `links`. This pull request patches this.

I've also included tests for staff privileges as part of the upcoming superuser/staff split work.

### CODE REVIEW

File: tests/sentry/web/frontend/test_react_page.py [hunk 1]
```diff
@@ -261,31 +261,71 @@ def test_handles_unknown_url_name(self):
...
+                assert response.redirect_chain == [
+                    (f"http://{other_org.slug}.testserver/issues/", 302)
+                ]
+            else:
+                assert response.redirect_chain == [
+                    (f"http://{other_org.slug}.testserver/auth/login/{other_org.slug}/", 302)
+                ]
+            assert self.client.session["activeorg"] == other_org.slug
 
             # Access another org as superuser on non-customer domain
```
Comment (markstory) [inline tests/sentry/web/frontend/test_react_page.py:310]: Does this comment need to change too?

File: tests/sentry/web/frontend/test_react_page.py [hunk 2]
```diff
@@ -261,31 +261,71 @@ def test_handles_unknown_url_name(self):
...
-            assert self.client.session["activeorg"] == org.slug
+            if is_superuser:
+                assert response.redirect_chain == [
+                    (f"http://{other_org.slug}.testserver/issues/", 302)
+                ]
+            else:
+                assert response.redirect_chain == [
+                    (f"http://{other_org.slug}.testserver/auth/login/{other_org.slug}/", 302)
+                ]
+            assert self.client.session["activeorg"] == other_org.slug
```
Comment (markstory) [inline tests/sentry/web/frontend/test_react_page.py:308]: For staff that don't have access to the organization should we be setting `activeorg` to the org they cannot access?

Comment (dashed) [reply tests/sentry/web/frontend/test_react_page.py:308]: @markstory Yes this is fine because we generally use `activeorg` to know which org slug to use when a user accesses `sentry.io` with no org slug given. 

If the user has no permissions, we just redirect them to the login page.

This is the same convention we use in the customer domain middleware here: https://github.com/getsentry/sentry/blob/f0ee34a650fe90733819f0294fccf5b1088b96c0/src/sentry/middleware/customer_domain.py#L107

**Review by markstory — APPROVED** [review]

File: src/sentry/web/client_config.py [hunk 3]
```diff
@@ -130,14 +146,26 @@ def _resolve_last_org(session, user, org_context=None):
...
             )
 
-    if org_context and org_context.member:
+    has_org_access = bool(org_context and org_context.member)
+
+    if not has_org_access and user_is_authenticated:
+        has_staff = features.has("auth:enterprise-staff-cookie", actor=user)
+        is_active_staff = has_staff and staff.is_active_staff(request)
+        is_active_superuser = not has_staff and superuser.is_active_superuser(request)
+        has_org_access = is_active_staff or is_active_superuser
```
Comment (cathteng) [inline src/sentry/web/client_config.py:155]: @schew2381 just to double check, do we need to check for staff here? superuser should be the only ones that can access customer orgs. staff should be able to hit a few endpoints that are also present in _admin

tl;dr i don't think this is necessary for staff -- we can just check for superuser and propogate the active org

Comment (schew2381) [reply src/sentry/web/client_config.py:155]: I think I'm lacking context, do staff need to have the active org on the session to access _admin? I'm not quite sure what "propagating the accessed org in links" means

Comment (cathteng) [reply src/sentry/web/client_config.py:155]: _admin is not tied to any org, which is why i think we don't have to check for staff here

Comment (schew2381) [reply src/sentry/web/client_config.py:155]: Hmm that makes sense, I think you're right then

Comment (schew2381) [reply src/sentry/web/client_config.py:155]: @dashed I think you can delete the staff part and just include a superuser check
